rkt/images.go

```diff
@@ -636,6 +713,8 @@ func (f *fetcher) downloadHTTP(url, label string, out writeSyncer, etag string)
 		return nil, fmt.Errorf("bad HTTP status code: %d", res.StatusCode)
```

Comment:
Added the logic for this. If `http.StatusRequestedRangeNotSatisfiable` is the HTTP code, the out writeSyncer is a file, and the file has a non-zero size, then we seek to the beginning of the file, truncate it to 0 bytes, and retry the download. Otherwise we `fallthrough` to the error case.